src/handlers/miner_dkg_penalty_handler.erl

```diff
@@ -0,0 +1,232 @@
+                           end, Members),
+    {ok, #state{bbas=maps:from_list(BBAs),
+                privkey=PrivKey,
+                f=F,
+                delay=Delay,
+                height=Height,
+                members=Members,
+                dkg_results=DKGResults}}.
+
+handle_command(start, State) ->
+    {NewBBAs, OutMsgs} = lists:foldl(fun({BBAIndex, SawSig}, {BBAs, Msgs}) ->
+                                             Input = case SawSig of
+                                                         true -> 1;
+                                                         false -> 0
+                                                     end,
+                                             {NewBBA, BBAMsgs} = hbbft_bba:input(maps:get(BBAIndex, State#state.bbas), Input),
+                                             {maps:put(BBAIndex, NewBBA, BBAs), fixup_bba_msgs(BBAMsgs, BBAIndex)++Msgs}
+                                     end, {State#state.bbas, []}, lists:zip(lists:seq(1, length(State#state.members)), State#state.dkg_results)),
+    {reply, ok, OutMsgs, State#state{bbas=NewBBAs}};
+handle_command(stop, State) ->
+    lager:info("stop called without timeout"),
+    {reply, ok, [{stop, 0}], State}.
+
+handle_message(BinMsg, Index, State) when is_binary(BinMsg) ->
+    try binary_to_term(BinMsg) of
+        Msg ->
+            handle_message(Msg, Index, State)
+    catch _:_ ->
+            lager:warning("got truncated message: ~p:", [BinMsg]),
+            ignore
+    end;
+handle_message({conf, Signatures}, _Index, State)
+  when State#state.artifact /= undefined andalso State#state.done_called == false ->
+    GoodSignatures =
+        lists:foldl(
+          fun({Address, Signature}, Acc) ->
+                  %% only check signatures from members we have not already verified and have not already appeared in this list
+                  case {lists:keymember(Address, 1, Acc) == false andalso
+                        lists:member(Address, State#state.members),
+                        blockchain_txn_consensus_group_failure_v1:verify_signature(State#state.artifact, Address, Signature)} of
+                      {true, true} ->
+                          lager:debug("adding sig from conf"),
+                          [{Address, Signature}|Acc];
+                      {true, false} ->
+                          lager:debug("got invalid signature ~p from ~p", [Signature, Address]),
+                          Acc;
+                      {false, _} ->
+                          Acc
+                  end
+          end, State#state.signatures, Signatures),
+        Majority = floor(length(State#state.members) / 2) + 1,
+    case length(GoodSignatures) of
+        SigLen when SigLen >= (State#state.f*2) + 1 ->
+            done(GoodSignatures, State#state.artifact),
+            {State#state{signatures=GoodSignatures, done_called=true}, []};
+        SigLen when SigLen >= Majority andalso State#state.conf_sent == false ->
```

Comment:
I'm wondering whether this node should multicast `{conf, GoodSigs}` even if it sent conf before when it sees `SigLen >= Majority`, mostly because it would help propagate conf msgs faster, I _think_. If I'm entirely mistaken, can we add a little comment here with some reasoning plz?